rocsolver/library/src/include/rocsolver_logger.hpp

```diff
@@ -200,19 +208,22 @@ class rocsolver_logger
         from_profile.level = from_stack.level;
         from_profile.calls++;
         from_profile.time += time;
+    }
 
-        rocsolver_logger::_mutex.unlock();
+    static std::unique_lock<std::mutex> acquire_lock()
+    {
+        return std::unique_lock<std::mutex>(rocsolver_logger::_mutex);
     }
 
 public:
     // return the singleton instance
-    static inline rocsolver_logger* instance()
+    static rocsolver_logger* instance()
     {
         return rocsolver_logger::_instance;
     }
 
     // returns true if logging facilities are enabled
-    static inline bool is_logging_enabled()
```

Comment:
I would at least like to keep this inline, since this function is called from every internal rocSOLVER routine even when logging has not been initialized or enabled.

Reply:
Any function defined within the class body is implicitly inline.

Reply:
Is there a benefit to not being explicit with this inline?

Reply:
I think it would get inlined anyways, but if you want to be sure, you can add the __force_inline / always_inline attribute. https://clang.llvm.org/docs/AttributeReference.html#always-inline-force-inline

Reply:
I want to be sure, yeah

Reply:
I'll swap this out for `__force_inline`